app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt

```diff
@@ -407,6 +410,26 @@ class UploadWorker(var appContext: Context, workerParams: WorkerParameters) :
                     contribution.state = Contribution.STATE_FAILED
                     contribution.chunkInfo = null
                     contributionDao.saveSynchronous(contribution)
+
+                    //If the Current Login is Invalid then ask the User to Login Again
+                    if(stashUploadResult.message.equals("Invalid token, or login failure.")){
+                        Timber.e("Invalid Login, logging out")
+                        sessionManager.logout()
+                            .andThen(Completable.fromAction {
+                                Timber.d("All accounts have been removed")
+                            })
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .subscribe(
+                                {
+                                    //Send Extra Login Message to LoginActivity
+                                    val intent = Intent(appContext, LoginActivity::class.java)
+                                    intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+                                    intent.putExtra("loginMessage", "Your Login has expired, Please Login Again.")
```

Comment:
This should use a resource string, so that it is translatable. Additionally I don't think `Login` needs to be capitalised.